app/src/main/java/com/mercadolibre/android/point_mainapp_demo/app/view/nfc/DetectCartActivity.kt

```diff
@@ -0,0 +1,153 @@
+        ActivityDetectCartBinding.inflate(layoutInflater)
+    }
+
+    private val nfcConfig = MPManager.nfcConfig
+
+    private var serialInfo = byteArrayOf()
+
+    private var operationCompleted = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+        initDetectionNfc()
+        setListeners()
+        binding.buttonNfcDetection.setOnClickListener {
+            operationClose()
+        }
+    }
+
+    private fun initDetectionNfc() {
+        setViews(DETECTING, "Detecting card NFC...")
+        nfcConfig.nfcOpenTools.open { mpResponse ->
+            mpResponse.doIfSuccess {
+                operationDetectNfc()
+            }.doIfError {
+                toast("Operation open: ${it.message}")
+            }
+        }
+    }
+
+    private fun operationDetectNfc() {
+        operationLedOn(NfcAvailableLed.LED_2)
+        nfcConfig.nfcDetectTools.detectCard(TIMEOUT) { mpResponse ->
+            mpResponse.doIfSuccess { result ->
+                setViews(
+                    SUCCESS,
+                    MESSAGE_DETECTION_DESCRIPTION.format(result.serialInfo.toString())
+                )
+            }.doIfError { exception ->
+                resultException(exception)
+            }
+        }
+    }
+
+    private fun operationLedOn(led: NfcAvailableLed) {
+        if (!operationCompleted) {
+            nfcConfig.nfcLedOnTools.ledOn(nfcAvailableLed = led) { mpResponse ->
+                onResultOperation(mpResponse, "Operation led on:")
+                operationCompleted = true
+            }
+        }
+    }
+
+    private fun onResultOperation(mpResponse: MPResponse<Boolean>, message: String) {
+        mpResponse.doIfSuccess {
+            toast("$message ${it}")
+        }.doIfError {
+            toast("$message ${it.message}")
+        }
+    }
+
+    private fun resultException(exception: Exception) {
+        if (exception is NfcTimeoutException) {
+            setViews("Timeout", exception.message.orEmpty())
+        } else {
+            setViews("Error", exception.message.orEmpty())
+        }
+    }
+
+    private fun setViews(title: String, description: String) {
+        binding.apply {
+            when (title) {
+                DETECTING -> {
+                    "$title\n$description".also { textNfcDetectionDescription.text = it }
+                }
+
+                SUCCESS -> {
+                    "$title\n$description".also { textNfcDetectionDescription.text = it }
+                    imageNfcDetectionCard.setImageResource(R.drawable.point_mainapp_demo_app_ic_check_white)
+                    cardViewNfcReader.visible()
+                    cardViewNfcWriter.visible()
+                    buttonNfcDetection.gone()
+                    operationLedOff()
+                }
+
+                else -> {
+                    "$title\n$description".also { textNfcDetectionDescription.text = it }
```

Comment:
Esto puede estar fuera del when? asi lo llamas una sola vez.

"$title\n$description".also { textNfcDetectionDescription.text = it }